packages/front-end/components/Experiment/ImportExperimentList.tsx

```diff
@@ -205,22 +206,19 @@ const ImportExperimentList: FC<{
                           onClick={(ev) => {
                             ev.preventDefault();
                             const importObj: Partial<ExperimentInterfaceStringDates> = {
-                              name: e.trackingKey,
+                              name: e.experimentName || e.trackingKey,
                               trackingKey: e.trackingKey,
                               datasource: data?.experiments?.datasource,
                               exposureQueryId: e.exposureQueryId || "",
-                              variations: e.variationKeys.map((v) => {
-                                const vInt = parseInt(v);
-                                const name = !Number.isNaN(vInt)
-                                  ? vInt == 0
-                                    ? "Control"
-                                    : `Variation ${vInt}`
-                                  : v;
+                              variations: e.variationKeys.map((vKey, i) => {
+                                const vName =
+                                  e.variationNames[i] ||
+                                  (i == 0 ? "Control" : `Variation ${i}`);
```

Comment:
This has a few problems:
1. The row order is random, so you can't rely on the first one always being the Control
2. If there's no name, but the variation id is a string, we should use the id instead of "Control" and "Variation X"
3. If the variation name is an integer like `0` or `1`, we should rename it to "Control" or "Variation X"

I think the following logic is simpler to understand and fixes all of the above issues:

```ts
variations: e.variationKeys.map((v) => {
  let name = e.variationNames[i] || v;
  // If the name is an integer, rename 0 to "Control" and anything else to "Variation {name}"
  if (name.match(/^[0-9]{1,2}$/)) {
    name = name === "0" ? "Control" : `Variation ${name}`;
  }
```

Reply:
Resolved by new changes